src/assisted_installer_controller/assisted_installer_controller.go

```diff
@@ -643,17 +639,17 @@ func (c controller) waitForOLMOperators() bool {
 
 // validateConsoleAvailability checks if the console operator is available
 func (c controller) validateConsoleAvailability() bool {
```

Comment:
`validateConsoleAvailability` now has an additional functionality.
It both validates the operator status locally and makes sure, in some cases, that it is up to date on `assisted-service` side.

It looks like it should be done in two different functions, in which the latter gets invoked upon need, for the `assisted-service` side validation.

However, I do see it is all nested under `WaitForPredicate`, and it might make it needlessly complex, so break it into two functions.

So the point that I'm making here is that from the outside, it is not clear `validateConsoleAvailability` makes that server-side validation, so can we at least add a docstring to elaborate that?

Reply:
I split the function into two functions: `isOperatorAvailableInCluster` and `isOperatorAvailableInService`